api/database/sampleApp.py

```diff
@@ -0,0 +1,68 @@
+from flask import Flask, request, jsonify
+from mongoengine import *
+from pymongo.errors import DuplicateKeyError
+from users import Users
+
+app = Flask(__name__)
+
+@app.route('/users', methods = ["GET"])
+def get_users():
+    name = request.args.get('Name')
+    email = request.args.get('Email')
+    password = request.args.get('Password')
+
+    data = []
+    for user in Users.objects:
+        if name is not None:
```

Comment:
Where does this Users object come from? Is it possible for a user to not have a name and/or email?

Reply:
Users.object is from the mongoengine. Name and email in this case is for searching ('GET'ing) only the users with the specified name or specified email. So if you don't specify a name then name can be None. If you don't specify name or email in the api call then this function returns a list of all users.